src/Processor/LastModifiedProcessor.php

```diff
@@ -8,28 +8,83 @@
+ * Set a "LastModified" property based on the last modified date set by the provider.
+ * E.g, for the {@see LocalFilesystemProvider}, the file mtime on the filesystem.
+ *
+ * If available, for local files, it'll use Git to get the last commit date for this file.
  */
 class LastModifiedProcessor implements ProcessorInterface
 {
     private string $property;
+    /** Git executable path on the system / PATH used to get the last commit date for the file, or null to disable. */
+    private ?string $gitPath;
+    private LoggerInterface $logger;
 
-    public function __construct(string $property = 'lastModified')
-    {
+    private static ?bool $gitAvailable = null;
+
+    public function __construct(
+        string $property = 'lastModified',
+        ?string $gitPath = 'git',
+        ?LoggerInterface $logger = null
+    ) {
         $this->property = $property;
+        $this->gitPath = $gitPath;
+        $this->logger = $logger ?? new NullLogger();
     }
 
     public function __invoke(array &$data, Content $content): void
     {
         if (\array_key_exists($this->property, $data)) {
-            // Last modified already set.
+            // Last modified already set (even if explicitly set as null).
             return;
         }
 
         $data[$this->property] = $content->getLastModified() ? $content->getLastModified()->format(\DateTimeInterface::RFC3339) : null;
+
+        if (LocalFilesystemProviderFactory::TYPE !== ($content->getMetadata()['provider'] ?? null)) {
+            // Won't attempt with a non local filesystem content.
+            return;
+        }
+
+        if (null === $this->gitPath || false === self::$gitAvailable) {
+            // Don't go further if the git command is not available or the git feature is disabled
+            return;
+        }
+
+        $filePath = $content->getMetadata()['path'];
+        $process = new Process([$this->gitPath, 'log', '-1', '--format=%cd', '--date=iso', $filePath]);
+        $process->run();
+
+        if (!$process->isSuccessful()) {
+            if (DIRECTORY_SEPARATOR === '\\' || 127 === $process->getExitCode()) {
+                // Exit code 127 means the command was not found.
+                // Not sure yet for windows, so let's ignore for now...
+                self::$gitAvailable = false;
+
+                $this->logger->info('Git was not found at path "{gitPath}". Check the binary path is correct or part of your PATH.', [
+                    'gitPath' => $this->gitPath,
+                    'output' => $process->getOutput(),
+                    'err_output' => $process->getErrorOutput(),
+                ]);
+
+                return;
+            }
+
+            // Throw on other errors:
+            throw new ProcessFailedException($process);
+        }
+
+        self::$gitAvailable = true;
+        $lastCommitDate = new \DateTime($process->getOutput());
+        $data[$this->property] = $lastCommitDate;
```

Comment:
Datetime instance seems more natural to me yes.

Reply:
Indeed, and it won't impose a format while denormalizing. We did format it before because we didn't have https://github.com/StenopePHP/Stenope/blob/master/src/Serializer/Normalizer/SkippingInstantiatedObjectDenormalizer.php yet, which ignore already instantiated objects while denormalizing.